src/chapters/databases/introduction.rst

```diff
@@ -0,0 +1,127 @@
+What are Databases?
+===================
+
+In the previous chapter, we learned how to use sessions to store data on a
+user's device. This allows us to make the information persist after refreshing
+a web page or stopping and restarting our Flask app. However, the data is still
+temporary. It will be lost when the user quits their browser or clears their
+cookies.
+
+In this chapter, we will learn how to store data in a more permanent way. We
+will also keep the data off of users' machines, which gives us greater control
+over the information.
+
+.. index:: ! database
+
+A **database** stores information and organizes it in a specific way. Unlike
+cookies, which contain a single key/value pair, databases allow us to manage a
+huge amount of content. Think of a database like a large hard drive. However,
+its only job is to store information. It runs no programs.
+
+Using a database instead of sessions lets developers decide what information to
+save, who can access it, and how long to keep it.
+
+Adding a Database
+-----------------
+
+Databases remain separate from both the web server and a user's computer. The
+diagram below shows how a database fits in with the server and a browser.
+
+.. figure:: figures/mvc-lite.png
+   :alt: Showing the interactions between a browser, a server program, a data management program, and a database.
+   :width: 80%
+
+The figure splits the task of running a website into three large pieces. Let's
+take a look at the different sections.
+
+#. A user launches their browser and navigates to a website. They send HTTP
+   requests to the web server, and their browser displays the responses on the
+   screen. The user can only interact with the server. 
+#. The database only interacts with a special program built to manage data. The
+   code accesses the database and creates new entries, reads information,
+   updates exiting data, or deletes content that is no longer needed. The
+   program does NOT communicate directly with visitors to the website.
+#. A *control* program sits in between. It receives HTTP requests and sends
+   back the responses. The code manages the web server, makes decisions, sends
+   data to the browser, and interacts with the data management program. The
+   control provides a useful middle step. It analyzes HTTP requests and filters
+   out anything that might corrupt the database or access it without permission.
+
+In the Flask applications we've built so far, ``main.py`` serves as the
+control.
+
+.. admonition:: Warning
+
+   News stories pop up frequently about hackers stealing information or
+   companies accidentally posting their customers' data online.
+   
+   When you attach a database to your web application, you MUST take steps to
+   protect the information. Visitors to your website should never have direct
+   access to where you store sensitive data.
+
```

Comment:
Really wanted a bit more info- maybe a link or suggestion for a google search for more info to students so inclined?

Reply:
I thought the same thing, but I couldn't find a good, accessible webpage for beginners.  If you have a suggestion, I'm all ears!

Reply:
I really liked this guys video: https://www.youtube.com/watch?v=LsHf3JSDBVc and this article was insightful: https://ckraczkowsky.medium.com/building-a-secure-admin-interface-with-flask-admin-and-flask-security-13ae81faa05

Reply:
As per our conversation, I added the standard "beyond the scope of this course" comment.